strawberryfields/io/xir_io.py

```diff
@@ -273,20 +315,29 @@ def to_xir(prog, **kwargs):
     return xir_prog
 
 
-def _listr(mixed_list):
-    """Casts a nested iterable to a list recursively"""
+def _listr(mixed_iterable: Iterable) -> List:
+    """Casts a nested iterable to a list recursively, maintaining the same shape.
+
+    Any iterable will be cast to a list, including casting all internal types to native Python
+    types (e.g., ``Decimal`` and ``np.floating`` to ``float``), maintaining the shape.
+    """
     list_ = []
 
-    for l in mixed_list:
-        if isinstance(l, Iterable):
+    for l in mixed_iterable:
+        # if string, then create a list of chars
+        if isinstance(l, str):
+            list_.append(l if len(l) == 1 else list(l))
```

Comment:
Is this actually desirable? Why not just write
```suggestion
        if isinstance(l, str):
            list_.append(l)
```

Reply:
Basically just to get `_listr("abc")` -> `["a", "b", "c"]` instead of `[["a"], ["b"], ["c"]]`, since each atomic part of a string is also always a string, which we don't really want encapsulated in lists.

Reply:
Right, but what I'm proposing is `_listr(["abc"]) → ["abc"]`; the current implementation gives `[["a", "b", "c"]]`.

Wouldn't you agree that the former is more useful and expected?

Reply:
Oh sorry, I understand now. I'm not sure, to be honest. I don't think strings are ever actually passed to this function, and I'm not sure there ever will be. It's mainly a matter of what would be expected by a user/developer using this function. 

Since a string is an `Iterable`, I think this function should change its type to `list`, e.g., `"abc"` -> `["a", "b", "c"]`. If that's the case, I also think `["abc"]` -> `[["a", "b", "c"]]` makes more sense.

As to which is more useful, I don't know.

Reply:
Removed support for passing strings directly to the function. That now raises an error.